### ISSUE TITLE
MiniupnpdNot always opening ports for Destiny 2 PC

### ISSUE THREAD

**Issue by azure-xiii** [openwrt/packages#16973]
Maintainer: @ldir-EDB0
Environment: Device - R7800 as well an IQRouter V2
OpenWrt Versions - Generic 19.07.2, Generic OpenWrt 18.06.8, Hynmans last 18.06, multiple Hynmans Master, and multiple Hynmans 19.07 releases.

Description:

Basically whenever I start up Destiny 2 on PC I have a about a 1 in 10 chance of UPNP (using luci-app-upnp) opening the required ports. I have IGDv1 enabled. This issue happens only on Openwrt (every version I have tried)

If I use the stock Netgear firmware or my ASUS router with stock firmware... it will open the port everytime. I know I can manually port forward but I have two people who play Destiny in the house and would like to use UPNP if possible. I really prefer using Openwrt due to CAKE

Thanks!

```

**Comment by stintel** [comment]
Version 2.2.3 solved the IGDv1 problem, you no longer need the -igdv1 variant.. Please test and close this issue if the problem is solved.

**Comment by azure-xiii** [comment]
@stintel I am planning on trying this tomorrow. However when I go to the packages.... it looks like there are two now. miniupnpd-iptables_2.2.3-2 & miniupnpd-nftables_2.2.3-2 What are the diffrences?

**Comment by stintel** [comment]
The iptables variant uses iptables backend, like firewall3; nftables variant uses nftables backend, like firewall4.

**Comment by azure-xiii** [comment]
@stintel Got the snapshot installed and configured with the new miniupnpd-nftables... However the issue still persists. You can easily check this on Windows 11 by running "Xbox Console Companion" and going to the network tab. It will run a test and come back as moderate. You can also watch in the LUCI gui that no ports are never opened.

**Comment by stintel** [comment]
> Got the snapshot installed and configured with the new miniupnpd-nftables

Are you using firewall4?

**Comment by azure-xiii** [comment]
@stintel Yes, it looks like firewall4 is what is installed by default on the snapshots. Is there some extra config I need to do?

**Comment by azure-xiii** [comment]
On the Stable releases with the old miniupnpd... I could atleast get it to open the ports every once ina awhile (25% of the time) With the new snapshot builds, I haven't seen it open ports once.

**Comment by azure-xiii** [comment]
So I went back to stable openwrt... installed the new 2.2.3 (Iptables version since stable is on the old firewall). And same issue. UPNP never works. If I use the the 2.2.1 version... it works sometimes. Keep in mind UPNP has been busted since during the 18.06 updates.

**Comment by stintel** [comment]
> @stintel Yes, it looks like firewall4 is what is installed by default on the snapshots. Is there some extra config I need to do?

At the time you wrote you installed a snapshot with miniupnpd-nftables, firewall4 had only just been made the default so there is no way a snaphost image could have been using it already.


2.2.3 defaults to IGDv1 when an MS client is detected, see https://github.com/miniupnp/miniupnp/issues/539. I guess you should start by checking if the user-agent is microsoft in Destiny's UPnP request. I only have Linux PCs so I can't test that, but my Xbox Series X works fine with 2.2.3, so the fix is definitely working.

**Comment by azure-xiii** [comment]
I did indeed update to the Jan 20th snapshot before checking the firewall. I would think that the xbox companion app on windows 10/11 would be using the MS client. I don't have an xbox to test it.

I tried a bunch of stuff last night and the only routers that trigger it are Stock Firmware Archer A7, stock firmware Asus AC86U, and my stock firmware Netgear R7800.

As soon as I try openwrt on any of the devices including an IQROUTER V2 I have... I cannot get upnp to work with any application on either of my PCs.

Not sure where to go from here. Thank you for your help though.

**Comment by stintel** [comment]
I would suggest you build miniupnpd with this change applied to the Makefile:

```diff
diff --git a/net/miniupnpd/Makefile b/net/miniupnpd/Makefile
index deeac1ccd..023b0f637 100644
--- a/net/miniupnpd/Makefile
+++ b/net/miniupnpd/Makefile
@@ -78,7 +78,8 @@ CONFIGURE_ARGS = \
        --leasefile \
        --portinuse \
        --firewall=$(BUILD_VARIANT) \
-       --disable-fork
+       --disable-fork \
+       --debug

 TARGET_CFLAGS += $(FPIC)
 TARGET_LDFLAGS += -Wl,--gc-sections,--as-needed
```
This will enable the following log line: https://github.com/miniupnp/miniupnp/blob/61059fcd0b724874f7b2d672249ddce96b3f607c/miniupnpd/upnphttp.c#L473

**Comment by azure-xiii** [comment]
Ok I will work on trying to build this. I haven't dug into building things out from openwrt yet.

**Comment by azure-xiii** [comment]
So I haven't been able to rebuild a package yet. Tried to do it in WSL but it appears I didn't set it up right. (I never use linux so having to learn all of this)

HOWEVER I found a igdv1 only build of miniupnpd on the forums. This 80% fixed Destiny 2(still doesn't open everytime) but did not fix Xbox Live on Windows 10/11 (can be tested from the settings menu under Gaming in win10. You need the Xbox Console Companion App for Win11)

Even if you just run the "Netsh INT Teredo Show Stat" command on windows... it says

NAT RESTRICTED 
UPNP NO

**Comment by tiagogaspar8** [comment]
Hi @azure-xiii, can you retest with the latest commit recently merged?
